galaxy_ng/app/api/base.py

```diff
@@ -1,36 +1,37 @@
-from django.conf import settings
+from django.test.signals import setting_changed
 
 from rest_framework import generics
 from rest_framework import views
 from rest_framework import viewsets
-from rest_framework.settings import perform_import
-
-
-GALAXY_EXCEPTION_HANDLER = perform_import(
-    settings.GALAXY_EXCEPTION_HANDLER,
-    'GALAXY_EXCEPTION_HANDLER'
-)
-GALAXY_AUTHENTICATION_CLASSES = perform_import(
-    settings.GALAXY_AUTHENTICATION_CLASSES,
-    'GALAXY_AUTHENTICATION_CLASSES'
-)
-GALAXY_PERMISSION_CLASSES = perform_import(
-    settings.GALAXY_PERMISSION_CLASSES,
-    'GALAXY_PERMISSION_CLASSES',
-)
-GALAXY_PAGINATION_CLASS = perform_import(
-    settings.GALAXY_PAGINATION_CLASS,
-    'GALAXY_PAGINATION_CLASS'
-)
+from rest_framework.settings import APISettings
+
+
+DEFAULT_SETTINGS = {
+    "GALAXY_EXCEPTION_HANDLER": "galaxy_ng.app.api.exceptions.exception_handler",
+    "GALAXY_PAGINATION_CLASS": "galaxy_ng.app.api.pagination.LimitOffsetPagination",
+    "GALAXY_AUTHENTICATION_CLASSES": [
+        "rest_framework.authentication.SessionAuthentication",
+        "rest_framework.authentication.TokenAuthentication",
+    ],
+    "GALAXY_PERMISSION_CLASSES": ["rest_framework.permissions.IsAuthenticated",],
+}
+IMPORT_STRINGS = [
+    "GALAXY_EXCEPTION_HANDLER",
+    "GALAXY_AUTHENTICATION_CLASSES",
+    "GALAXY_PERMISSION_CLASSES",
+    "GALAXY_PAGINATION_CLASS",
+]
+
+local_settings = APISettings(defaults=DEFAULT_SETTINGS, import_strings=IMPORT_STRINGS)
```

Comment:
I'm a little concerned about changing the way the base view uses settings (mostly because of the external use warning mentioned above). But I don't really know a better approach.

Could the tests run with a different DJANGO_SETTINGS_MODULE that would always defaults to the standalone settings?  Or do the tests need to switch back and forth at runtime?

Reply:
I share your concern. Probably we should find a better workaround for this. Honestly haven't noticed that warning. For tests only one could use mocks to override settings in LocalSettingsMixin. Otherwise implementing some home grown alternative to the API settings.

The key problem about LocalSettingsMixin is that it's properties resolved at the import time, so it's not enough to use `@override_settings` decorator. Which I think may be even better.